neurom/core/_neuron.py

```diff
@@ -28,16 +28,176 @@
 
 """Neuron classes and functions."""
 
-from copy import deepcopy
+from collections import deque
 from itertools import chain
 
+import morphio
 import numpy as np
 from neurom import morphmath
-from neurom.core._soma import Soma
+from neurom.core._soma import make_soma
 from neurom.core.dataformat import COLS
-from neurom.utils import memoize
+from neurom.core.types import NeuriteIter, NeuriteType
+
+
+class Section:
+    """Simple recursive tree class."""
+
+    def __init__(self, morphio_section):
+        """The section constructor."""
+        self.morphio_section = morphio_section
+
+    @property
+    def id(self):
+        """Returns the section ID."""
+        return self.morphio_section.id
+
+    @property
+    def parent(self):
+        """Returns the parent section if non root section else None."""
+        return None if self.morphio_section.is_root else Section(self.morphio_section.parent)
```

Comment:
for this kind of function I usually avoid ternary op: 
```
if self.morphio_section.is_root:
     return None
return Section(self.morphio_section.parent)
```

Reply:
done